stackslib/src/config/mod.rs

```diff
@@ -1210,53 +1210,321 @@ impl std::default::Default for Config {
+    /// This is intended strictly for testing purposes.
+    ///
+    /// Default: `None` (uses the burnchain's default starting height for the mode).
     pub first_burn_block_height: Option<u64>,
+    /// Overrides the default starting block timestamp of the burnchain.
+    /// Should be used together with [`BurnchainConfig::first_burn_block_height`] and [`BurnchainConfig::first_burn_block_hash`] for proper operation.
+    ///
+    /// Applied only if [`BurnchainConfig::mode`] is not "mainnet".
+    /// This is intended strictly for testing purposes.
+    ///
+    /// Default: `None` (uses the burnchain's default starting timestamp).
     pub first_burn_block_timestamp: Option<u32>,
+    /// Overrides the default starting block hash of the burnchain.
+    /// Should be used together with [`BurnchainConfig::first_burn_block_timestamp`] and [`BurnchainConfig::first_burn_block_height`] for proper operation.
+    ///
+    /// Applied only if [`BurnchainConfig::mode`] is not "mainnet".
+    /// This is intended strictly for testing purposes.
+    ///
+    /// Default: `None` (uses the burnchain's default starting block hash).
     pub first_burn_block_hash: Option<String>,
-    /// Custom override for the definitions of the epochs. This will only be applied for testnet and
-    /// regtest nodes.
+    /// Custom override for the definitions of Stacks epochs (start/end burnchain heights, consensus rules).
+    /// This setting allows testing specific epoch transitions or custom consensus rules by defining exactly
+    /// when each epoch starts on bitcoin.
+    ///
+    /// **Configuration:**
+    /// Configured as a list `[[burnchain.epochs]]` in TOML, each with:
+    /// - `epoch_name`: A string identifying the epoch (e.g., `"1.0"`, `"2.05"`, `"3.1"`).
+    /// - `start_height`: The Bitcoin block height at which this epoch becomes active.
+    ///
+    /// Example TOML entry:
+    /// ```toml
+    /// [[burnchain.epochs]]
+    /// epoch_name = "2.1"
+    /// start_height = 150
+    /// ```
+    ///
+    /// **Details:**
+    /// Epochs define distinct protocol rule sets (consensus rules, execution costs, capabilities).
+    /// When configured, the list must include all epochs sequentially from "1.0" up to the
+    /// highest desired epoch, without skipping any intermediate ones.
+    /// Valid `epoch_name` values currently include:
+    /// `"1.0"`, `"2.0"`, `"2.05"`, `"2.1"`, `"2.2"`, `"2.3"`, `"2.4"`, `"2.5"`, `"3.0"`, `"3.1"`.
+    ///
+    /// **Validation Rules:**
+    /// - Epochs must be provided in strict chronological order (`1.0`, `2.0`, `2.05`...).
+    /// - `start_height` values must be non-decreasing across the list.
+    /// - Epoch `"1.0"` must have `start_height = 0`.
+    /// - The number of defined epochs cannot exceed the maximum supported by the node software.
+    ///
+    /// Applied only if [`BurnchainConfig::mode`] is not "mainnet".
+    /// This is intended strictly for testing purposes.
```

Comment:
Inconsistent use of this line. for example, `pox_reward_length` and `pox_2_activation` should probably also have this line (as well as a few others in this PR). 
I think it should be present on any setting that is used for testing only